File: cpp/1D_DP/198.house-robber.cpp
#include "../std.hpp"
class Solution {
public:
    int rob(vector<int> &nums)
    {
        int prev = 0, curr = 0, next = 0;
        for (auto &&n : nums) {
            next = max(prev + n, curr);
            prev = curr;
            curr = next;
        }
        return curr;
    }
};

File: js/1D_DP/198.house-robber.ts
function rob(nums: number[]): number {
  let rob1 = 0,
    rob2 = 0
  for (let num of nums) {
    let newRob = Math.max(rob1 + num, rob2)
    rob1 = rob2
    rob2 = newRob
  }
  return rob2
}
